bin/varnishd/cache/cache_fetch.c

```diff
@@ -617,6 +617,9 @@ vbf_stp_fetchbody(struct worker *wrk, struct busyobj *bo)
 		}
 	} while (vfps == VFP_OK);
 
+	if (!vfc->failed && bo->htc->body_status == BS_TRAILERS)
+		vfc->failed = VDI_GetTrl(bo);
```

Comment:
Should this be squashed into the commit that causes connection pooling to break with backend chunked responses?

Reply:
I don't have time to look closer, because this change is awaiting other h2-related work, but we'll look at this when it's time to rebase the branch.